File: app/src/main/java/linusfessler/alarmtiles/alarm/AlarmRepository.java
package linusfessler.alarmtiles.alarm;

import androidx.lifecycle.LiveData;

import java.util.concurrent.ExecutorService;

import javax.inject.Inject;
import javax.inject.Singleton;

import linusfessler.alarmtiles.core.AppDatabase;
import lombok.AccessLevel;
import lombok.Getter;

@Singleton
class AlarmRepository {

    private final ExecutorService writeExecutor;
    private final AlarmDao alarmDao;

    @Getter(AccessLevel.PACKAGE)
    private final LiveData<Alarm> alarm;

    @Inject
    AlarmRepository(final AppDatabase appDatabase) {
        writeExecutor = appDatabase.getWriteExecutor();
        alarmDao = appDatabase.alarmDao();

        alarm = alarmDao.select();
    }

    void update(final Alarm alarm) {
        writeExecutor.execute(() -> alarmDao.update(alarm));
    }
}

File: app/src/main/java/linusfessler/alarmtiles/shared/alarm/AlarmRepository.kt
package linusfessler.alarmtiles.shared.alarm

import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AlarmRepository @Inject constructor(database: AlarmDatabase) {
    private val dao = database.alarmDao()

    fun select(id: Long) = dao.select(id)

    fun insert(alarm: Alarm) {
        GlobalScope.launch {
            dao.insert(alarm)
        }
    }

    fun update(alarm: Alarm) {
        GlobalScope.launch {
            dao.update(alarm)
        }
    }
}